app/state_machines/configurable_state_machine/machine.rb

```diff
@@ -132,8 +133,10 @@ def extract_roles_from_metadata(metadata)
       user = extract_user_from_metadata(metadata)
       if team.nil?
         user.roles
-      else
+      elsif user.roles_for_team(team).any?
         user.roles_for_team(team).map(&:role)
+      else
```

Comment:
This seems dangerous. Should this branch be taken only when `user.nil?` perhaps?

Reply:
No.  It's taken because the tests don't always match a user with the correct role in the team.  It was you who said the state machine shouldn't be checking this.